os/dev/ext-flash.c

```diff
@@ -437,19 +483,31 @@ ext_flash_erase(size_t offset, size_t length)
 bool
 ext_flash_test(void)
 {
-  bool ret;
+  flash_spi_configuration = &flash_spi_configuration_default;
+
+  if(ext_flash_open() == false) {
+    return false;
+  }
 
-  ret = ext_flash_open();
-  ext_flash_close();
+  if(ext_flash_close() == false) {
+    return false;
+  }
 
-  return ret;
+  LOG_INFO("Flash test successful\n");
+
+  return true;
 }
 /*---------------------------------------------------------------------------*/
-void
-ext_flash_init()
+bool
+ext_flash_init(spi_device_t *conf)
```

Comment:
Maybe all functions but init could make use of the spi_device_t pointer as that would allow using the driver for multiple chips? (e.g. two external memories on the same PCB with one chip-select each?). Now if you have multiple chips you first have to call init and then read / write data. Not sure that it is anything critical but the dev structs is kind of intended for uses like that. ext_flash_open(my_dev); ext_flash_write(my_dev, offset, len, databuf); etc...

Reply:
Good point.

Reply:
This is implemented in the latest version below. Yet, the init function also takes as input the configuration pointer. This is because it initialises the part rather than the driver.